### PR TITLE
revert back to running latest blackduck detect 6.9.0

### PR DESCRIPTION
**Pull request by dasormeter**
We reverted daml repo back to prior 6.1.0 as that was main had been running on for some time and it continued to work.

This PR brings `daml` to work with the latest 6.9.0 version of detect.

The tl/dr is that it looks like default directory scanning and signature matching is more aggressive with 6.9.0 so it tried to recurse the world and ended up with too much to make sense of due to the >50gb nature of all the files under the daml repo with bazel and nix and friends.

This PR breaks the scan to detection by language, so that failures are easier to see and reason about, explicitly excludes various bazel related directories that are recursive onto each other and are of unreasonable size, and does not include the signature scanning, which recurses every binary and compares its signature to match in blackduck kb.

The remaining improvements / due dilligence to address later include

1. Make sure the built release artifacts (really just JARs at present) have dependencies and versions that exactly match up to what bazel tells us via `bazel deps....` query.  This would require running the release script and then performing a blackduck signature scan against those jars, and blackduck will do a signature match on each class file to determine its identity and version
2. The C libraries that are brought in by DAML components are not declared via bazel deps in a way we can make sense of, nor are they scannable or identifiable to blackduck.  These change rarely and have been vetted previously, but as an improvement we should see if it is possible to dynamically discover them and extract the metadata in a CLANG package format that blackduck can discover so that these are included in the bill of materials
3. Broad signature scanning -- we really care very little about Bazel and Nix as these are supporting dev tools, and we do not need to enumerate each and every library they include and their license or security vulnerability risk.  However, if there are libraries that we either include to run in our own CI infrastructure or that somehow make it into code that we distribute, we do want to do a signature match on this.

`6.9.0` docs https://community.synopsys.com/s/document-item?bundleId=integrations-detect&topicId=introduction.html&_LANG=enus

`6.1.0` docs https://blackducksoftware.github.io/synopsys-detect/6.1.0/

### CODE REVIEW

**Comment by dasormeter** [comment]
looks like this is still getting stuck in the scan

File: ci/cron/daily-compat.yml [hunk 1]
```diff
@@ -391,6 +391,8 @@ jobs:
       - bash: |
           set -euo pipefail
           eval "$(dev-env/bin/dade-assist)"
+          
+          export SCAN_CLI_OPTS="-Xms1g -Xmx10g"
```
Comment (cocreature) [inline ci/cron/daily-compat.yml:395]: shouldn’t we set this for both blackduck steps? You only set it for the second one.

Comment (dasormeter) [reply ci/cron/daily-compat.yml:395]: probably but the first haskell step seems to be fine so i just targeted it for the problematic one -- no harm in keeping it consistent for both though if this works

File: ci/cron/daily-compat.yml [hunk 2]
```diff
@@ -385,35 +385,89 @@ jobs:
...
-          --detect.excluded.directories=.bazel-cache,language-support/ts/codegen/tests/ts,language-support/ts,language-support/scala/examples/iou-no-codegen,language-support/scala/examples/quickstart-scala,docs/source/app-dev/bindings-java/code-snippets,docs/source/app-dev/bindings-java/quickstart/template-root,language-support/scala/examples/quickstart-scala,language-support/scala/examples/iou-no-codegen \
-          --detect.cleanup=false \
-          --detect.policy.check.fail.on.severities=MAJOR,CRITICAL,BLOCKER \
           --detect.notices.report=true \
-          --detect.cleanup.bdio.files=true \
           --detect.code.location.name=digital-asset_daml_${BAZEL_DEPENDENCY_TYPE} \
-          --detect.timeout=4500
-        displayName: 'Blackduck Scan'
+          --detect.timeout=1500
+        displayName: 'Blackduck Bazel Java Scala Scan'
```
Comment (cocreature) [inline ci/cron/daily-compat.yml:409]:
```suggestion
        displayName: 'Blackduck Bazel JVM Scan'
```

File: ci/cron/daily-compat.yml [hunk 3]
```diff
@@ -385,35 +385,89 @@ jobs:
...
+        env:
+          BLACKDUCK_HUBDETECT_TOKEN: $(BLACKDUCK_HUBDETECT_TOKEN)
+      - bash: |
+          set -euo pipefail
+          eval "$(dev-env/bin/dade-assist)"
+
+          cd language-support/ts && yarn install && cd ../../
```
Comment (cocreature) [inline ci/cron/daily-compat.yml:416]:
```suggestion
          (cd language-support/ts && yarn install)
```

**Review by cocreature — APPROVED** [review]
Thanks for pushing this through!

**Pull request merged**